clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1392,6 +1392,22 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
     return handleSpecialType(FD, FieldTy);
   }
 
+  RecordDecl *wrapPointer(FieldDecl *Field, QualType FieldTy) {
+    RecordDecl *WrapperClass =
+        SemaRef.getASTContext().buildImplicitRecord("wrapper_class");
+    WrapperClass->startDefinition();
+    Field = FieldDecl::Create(
+        SemaRef.getASTContext(), WrapperClass, SourceLocation(),
+        SourceLocation(), /*Id=*/nullptr, FieldTy,
```

Comment:
We probably want to make sure the pointer is in the 'global' namespace, right? Otherwise the runtime won't treat it as a USM, right?

Reply:
> We probably want to make sure the pointer is in the 'global' namespace, right?
This is already done by struct decomposition at this point.